pkg/container/docker.go

```diff
@@ -184,7 +199,20 @@ func (f *BuilderFactory) GenerateImageHash() (string, error) {
 		return "", ErrHashingBuildContext.Wrap(err)
 	}
 
-	logrus.Debug("Generated image hash: ", fmt.Sprintf("%x", hasher.Sum(nil)))
+	f.logger.Debug("Generated image hash: ", fmt.Sprintf("%x", hasher.Sum(nil)))
 
 	return fmt.Sprintf("%x", hasher.Sum(nil)), nil
 }
+
+func verifyOptions(opts BuilderFactoryOptions) error {
+	if opts.ImageName == "" {
```

Comment:
Is image name really required? Feels like we could just auto generate one on the fly.

Reply:
yes we need to have it and the generated image (i.e. built image)  will use that name. it is essential for the long running tests to have static names for all resources including images.